src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -417,10 +424,52 @@ void Terminal::UpdateAppearance(const ICoreAppearance& appearance)
     // before, and shouldn't be now either.
     _scrollOffset = originalOffsetWasZero ? 0 : static_cast<int>(::base::ClampSub(_mutableViewport.Top(), newVisibleTop));
 
+    // Now that we've finished the hard work of resizing the main buffer and
+    // getting the viewport back into the right spot, we need to ALSO resize the
+    // alt buffer, if one exists. Fortunately, this is easy. We don't need to
+    // worry about the viewport and scrollback at all! The alt buffer never has
+    // any scrollback, so we just need to resize it and presto, we're done.
+    if (_inAltBuffer())
+    {
+        _altBuffer->GetCursor().StartDeferDrawing();
+        // we're capturing `this` here because when we exit, we want to EndDefer on the (newly created) active buffer.
+        auto endDefer = wil::scope_exit([this]() noexcept { _altBuffer->GetCursor().EndDeferDrawing(); });
+
+        // First allocate a new text buffer to take the place of the current one.
+        std::unique_ptr<TextBuffer> newTextBuffer;
+        try
+        {
+            // GH#3848 - Stash away the current attributes
+            const auto oldBufferAttributes = _mainBuffer->GetCurrentAttributes();

```

Comment:
Is this right? We're resizing the alt buffer here, so why are we saving the main buffer attributes?

Reply:
no it DEFINITELY ISN'T! Good eye. More copypasta issues 😒